octobot_trading/exchanges/parser/util.py

```diff
@@ -0,0 +1,424 @@
+            except KeyError:
+                pass
+            try:
+                if use_info_sub_dict:
+                    value = self.raw_record[ExchangePositionCCXTColumns.INFO.value][key]
+                    if value:
+                        return value
+                    if value in allowed_falsely_values:
+                        potential_value = value
+            except KeyError:
+                pass
+        if potential_value is not None:
+            return potential_value
+        if not_found_val is not None:
+            return not_found_val
+
+        if enable_log:
+            self._log_missing(
+                key_to_set=key_to_set,
+                tried_message=keys_to_test,
+            )
+        return None
+
+    def _ensure_dict(self, raw_record):
+        if type(raw_record) is dict:
+            self.formatted_record = {}  # clear record first
+            self.raw_record = raw_record
+        else:
+            raise NotImplementedError(
+                self.debugging_report_template(
+                    f"{self.PARSER_TITLE} parser received an invalid format\n"
+                    "type should be a dict\n"
+                    f"received raw data type:  {type(raw_record) or 'no data type'}\n"
+                    f"received raw data:  {raw_record or 'no data'}\n"
+                )
+            )
+
+    def _ensure_list(self, raw_records):
+        if type(raw_records) is list:
+            self.raw_records = raw_records
+            self.multiple_records = True
+        else:
+            raise NotImplementedError(
+                self.debugging_report_template(
+                    f"{self.PARSER_TITLE} parser received an invalid format\n"
+                    f"type should be a list of multiple {self.PARSER_TITLE}\n"
+                    f"received raw data type:  {type(raw_records) or 'no raw data type'}\n"
+                    f"received raw data: {raw_records or 'no raw data'}"
+                )
+            )
+
+    def _log_missing(
+        self,
+        key_to_set: str,
+        tried_message: str,
+        additional_message: str = "",
+        error=None,
+    ):
+        _message = (
+            f"> Failed to parse {self.PARSER_TITLE} attribute: {key_to_set} - tried: {tried_message}\n"
+            f"{additional_message}{' - Error: '+repr(error) if error else ''}\n"
+        )
+        if key_to_set in self.debugging_report_dict:
+            self.debugging_report_dict[key_to_set].append(_message)
+        else:
+            self.debugging_report_dict[key_to_set] = [_message]
+
+    def _log_missing_with_method(self, key_to_set, tried_message, method, error):
+        self._log_missing(
+            key_to_set=key_to_set,
+            tried_message=tried_message,
+            additional_message=f"with method: {type(method).__name__ if method else 'no method provided'} - error: {repr(error)}\n",
+        )
+
+
+def convert_any_time_to_seconds(raw_time):
```

Comment:
Why not two separate methods? Because your first check is a bit "random" and in most cases we know what is the exchange date format

Reply:
I separated them and improved it, now its always going with standard first